### PR TITLE
fix(discovery): disclose featured tab sponsors

### PR DESCRIPTION
**Pull request by NotThatKindOfDrLiz**
## Summary

- Parse featured tab `pill_label` separately from sponsor disclosure, and resolve `disclosure_label` as `sponsorName` instead of tab chrome copy.
- Render the Featured tab label from local i18n, show the collection pill from `pill_label`, and show the exact paid-partnership disclosure above the featured feed when a sponsor is present.
- Add locale coverage and focused regression tests for parsing, eligibility, tab rendering, sponsor non-duplication, and empty/failed feed states.

> Avoid naming corporate partners, customers, or other sensitive external brands in this PR title or body. Use generic descriptors unless a maintainer explicitly approves the public reference.

## Motivation

- Keeps web aligned with the upcoming featured-tab sponsorship contract so a sponsor name does not replace the collection pill or get read as generic disclosure copy.
- Leaves placement hard-wiring for a separate PR so this change stays scoped to sponsorship semantics.

## Related Issue

- Refs #608

## Testing

- [x] `npm run test`
- [ ] Manual verification completed; production currently serves no live featured tabs to exercise manually.

## Visuals

- [x] UI change; no screenshot attached because production currently serves no live featured tabs.
- [x] Visuals and text avoid sensitive external brand or partner names unless explicitly approved

### CODE REVIEW

File: src/pages/DiscoveryPage.tsx [hunk 1]
```diff
@@ -252,21 +253,21 @@ export function DiscoveryPage() {
...
             style={{ gridTemplateColumns: `repeat(${tabItems.length}, minmax(0, 1fr))` }}
           >
-            {tabItems.map(({ value, label, Icon, disclosureLabel }) => (
+            {tabItems.map(({ value, label, Icon, pillLabel }) => (
               <TabsTrigger
                 key={value}
                 value={value}
                 className="min-w-0 gap-1.5 px-2 sm:gap-2 sm:px-4"
-                aria-label={disclosureLabel ? `${label}: ${disclosureLabel}` : label}
+                aria-label={label}
```
Comment (mbradley) [inline src/pages/DiscoveryPage.tsx]: The pill (`pillLabel`) is shown visually but `aria-label={label}` overrides the button's text content, so screen-reader users don't get it. #608 point 3 calls for the pill to carry a spoken label. Folding it into the accessible name:

```suggestion
                aria-label={pillLabel ? `${label}: ${pillLabel}` : label}
```

**Review by mbradley — APPROVED** [review]
Matches #608: the disclosure renders the signed-off "In paid partnership with {sponsor}" string exactly, and the tab label moves to a local i18n key. Placement and the player-surface disclosure are correctly left to follow-ups per the issue. One non-blocking a11y note below.
